filters/circuit/breaker.go

```diff
@@ -91,99 +68,30 @@ func (s *spec) Name() string {
 }
 
 func consecutiveFilter(args []interface{}) (filters.Filter, error) {
-	if len(args) == 0 || len(args) > 4 {
-		return nil, filters.ErrInvalidFilterParameters
-	}
-
-	failures, err := getIntArg(args[0])
-	if err != nil {
-		return nil, err
-	}
-
-	var timeout time.Duration
-	if len(args) > 1 {
-		timeout, err = getDurationArg(args[1])
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	var halfOpenRequests int
-	if len(args) > 2 {
-		halfOpenRequests, err = getIntArg(args[2])
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	var idleTTL time.Duration
-	if len(args) > 3 {
-		idleTTL, err = getDurationArg(args[3])
-		if err != nil {
-			return nil, err
-		}
-	}
-
+	a := filters.Args(args)
 	return &filter{
 		settings: circuit.BreakerSettings{
 			Type:             circuit.ConsecutiveFailures,
-			Failures:         failures,
-			Timeout:          timeout,
-			HalfOpenRequests: halfOpenRequests,
-			IdleTTL:          idleTTL,
+			Failures:         a.Int(),
+			Timeout:          a.DurationOr(0),
+			HalfOpenRequests: a.IntOr(0),
+			IdleTTL:          a.DurationOr(0),
```

Comment:
Optional arguments: converted if present otherwise default value is returned

Reply:
There can be more cases. The variadic number of arguments come to my mind. In those cases the last call could return a slice.

Reply:
>  last call could return a slice

There is Strings() method for that that return the rest of args converted to string slice.